components/conversations/conversation.controllers.js

```diff
@@ -0,0 +1,49 @@
+import Conversation from "./conversation.model.js";
+
+export const getConversation = async (req, res) => {
+  const { placeId, userId } = req.query;
+
+  try {
+    const result = await Conversation.findOne({ placeId, userId });
+    if (result) return res.send(result);
+
+    res.status(404).send();
+  } catch (err) {
+    res.status(500).send({ message: err });
+  }
+};
+
+export const toggleUser = async (req, res) => {
+  const { userId, placeId } = req.params;
+  try {
+    await Conversation.findOneAndUpdate({ userId, placeId }, [
+      { $set: { isActive: { $eq: [false, "$isActive"] } } },
+    ]);
+    res.send();
+  } catch (message) {
+    res.status(500).send({ message });
+  }
+};
+
+export const getAllConversations = async (req, res) => {
+  const { placeId, userId } = req.query;
+  let result;
+  try {
+    if (placeId)
+      result = await Conversation.find(
```

Comment:
u can use the query once and save user/place id in variable by using or operator:

const queryBy=placeId || userId
const result = await Conversation.find({ queryBy });